packages/web-workers/src/types.d.ts

```diff
@@ -0,0 +1,17 @@
+export function createWorker<T>(load: () => Promise<T>): () => Promisify<T>;
+
+type Arguments<T> = T extends (...arg: infer U) => any ? U : any;
+
+export type Promisify<T> = T extends (...args: infer Args) => infer TypeReturned
+  ? TypeReturned extends () => infer Result
+    ? (...args: Args) => () => Promise<Result>
+    : TypeReturned extends Promise<any> // check to prevent returning Promise<Promise<thing>>> in async function
+      ? (...args: Args) => TypeReturned
+      : (...args: Args) => Promise<TypeReturned>
+  : T extends Array<infer ArrayElement>
+    ? ArrayOfPromises<ArrayElement>
+    : {[K in keyof T]: Promisify<T[K]>};
+
+interface ArrayOfPromises<T> extends Array<Promisify<T>> {}
+
+export type WorkerInput<T> = Promisify<T>;
```

Comment:
`WorkerInput` is not the same as `Promisify`. A worker **can** accept `() => string | Promise<string>`, so we should turn `() => string` into that instead of forcing a promise. A worker **can't** return a `() => string`, because that is returned over the bridge, which means it will only ever return a promise. So, `() => string` => `() => Promise<string>`.